paperqa/types.py

```diff
@@ -502,10 +515,15 @@ def overwrite_docname_dockey_for_compatibility_w_doc(
     ) -> dict[str, Any]:
         """Overwrite fields from metadata if specified."""
         overwrite_fields = {"key": "docname", "doc_id": "dockey"}
-        if data.get("overwrite_fields_from_metadata", True):
-            for field, old_field in overwrite_fields.items():
-                if data.get(field):
-                    data[old_field] = data[field]
+        for field, old_field in overwrite_fields.items():
```

Comment:
Can we have this be `DEFAULT_FIELDS_TO_OVERWRITE_FROM_METADATA`, and remove `"citation"` from `DEFAULT_FIELDS_TO_OVERWRITE_FROM_METADATA`

Reply:
The problem is that many test rely on the fact that `"citation"` is in that set

Reply:
Can't those tests then specify `DEFAULT_FIELDS_TO_OVERWRITE_FROM_METADATA | {"citation"}` into the `Settings` then?

Reply:
We can use an intersection...

Reply:
Yes but I think we want to have the citation overwritten by default.
We are replacing the `overwrite_fields_from_metadata` here and that flag was used to overwrite the `citation`.
I mean, this would be changing a default behavior.

Reply:
Other approach would be

```python
"""Overwrite fields from metadata if specified."""
overwrite_fields = {"key": "docname", "doc_id": "dockey"}
fields_to_overwrite = data.get(
    "fields_to_overwrite_from_metadata",
    DEFAULT_FIELDS_TO_OVERWRITE_FROM_METADATA,
)
for field in (overwrite_fields.keys() & fields_to_overwrite):
    if data.get(field):
        data[overwrite_fields[field]] = data[field]
return data
```

I think this makes sense

Reply:
Maybe I am not understanding, but why do we want to have the citation overwritten by default? Isn't that like a RAG QA manifest specific thing?

I think the best default for users is to infer citation

Reply:
Okay we talked in-person, I understand what you're saying here. Defer to your best judgement here, thanks for talking me through it 👌